### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by Juakata**
Add operate and calculate logic, I also add a test to avoid linter errors.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import operate from './operate';
+
+const calculate = (data, name) => {
```
Comment (azdrenymeri) [inline src/logic/calculate.js:3]: What happens if `name` equals `"AC"` or `"."` or `"="`?
Kindly cover all the cases.

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import Big from 'big-js';
+
+const operate = (n1, n2, operation) => {
```
Comment (azdrenymeri) [inline src/logic/operate.js:3]: What happens if there is a division by zero?

**Review by azdrenymeri — CHANGES_REQUESTED** [review]
### Hello @Juakata 
- [x] Improve your `calculate` function so when user presses `.`,`AC` or `=` what should be returned.
- [x] Also error handling on `operate.js` should be handled(divided by zero).
### Kindly implement these two fixes and add it for another review.

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      break;
+    case '.':
+      result = '.';
+      break;
+    case 'AC':
+      result = '';
+      break;
+    case '=':
+      result = operate(data[0], data[1], data[2]);
+      break;
```
Comment (shloch) [inline src/logic/calculate.js]: the way you using your `data` (data[0], data[1], data[2]), seems to be an array,
This should be an **OBJECT** with three properties {[see requiredments](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png)}

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    case '-':
+      result = x.minus(y);
+      break;
+    case '*':
+      result = x.times(y);
+      break;
+    case '/':
+      if (y === 0) {
+        return 'Math Error';
+      }
```
Comment (shloch) [inline src/logic/operate.js:20]: great

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @Juakata 
We are progressing, one more correction below :

--> calculator data object contains three properties: `total`, `next` and `operation`

Make sure thats an object with the above mentioned properties

### waiting for last review hopefully 😊😊

**Review by shloch — APPROVED** [review]
accomplished

## GOOD TO GO

HAppy coding 😊 😊 😊

**Pull request merged**